core/src/main/scala/org/genivi/sota/core/daemon/DeltaListener.scala

```diff
@@ -0,0 +1,61 @@
+package org.genivi.sota.core.daemon
+
+import akka.Done
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.core.UpdateService
+import org.genivi.sota.core.campaigns.CampaignLauncher
+import org.genivi.sota.core.db.Campaigns
+import org.genivi.sota.data.PackageId
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages.{GeneratedDelta, GeneratingDeltaFailed}
+import slick.driver.MySQLDriver.api._
+import eu.timepit.refined._
+import org.genivi.sota.marshalling.{DeserializationException, RefinementError}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeltaListener(deviceRegistry: DeviceRegistry, updateService: UpdateService,
+                    messageBus: MessageBusPublisher)(implicit db: Database)  {
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  def generatedDeltaAction(msg: GeneratedDelta)(implicit ec: ExecutionContext): Future[Done] = {
+    val fromR = refineV[PackageId.ValidVersion](msg.from) match {
+                  case Left(e) => throw DeserializationException(RefinementError(msg.from, e))
+                  case Right(v) => v
+                }
+    val toR = refineV[PackageId.ValidVersion](msg.from) match {
+                case Left(e) => throw DeserializationException(RefinementError(msg.from, e))
+                case Right(v) => v
+              }
+    val f = for {
+      id <- Campaigns.getIdFromDelta(fromR, toR)
+      _  <- Campaigns.setSize(id, msg.size)
+      lc <- Campaigns.fetchLaunchCampaignRequest(id)
+    } yield (id, lc)
+
+    db.run(f.transactionally).flatMap { case (id, lc) =>
+      CampaignLauncher.launch(deviceRegistry, updateService, id, lc, messageBus)(db, ec).map(_ => Done)
+    }
+  }
+
+  def deltaGenerationFailedAction(msg: GeneratingDeltaFailed)(implicit ec: ExecutionContext): Future[Done] = {
+    val fromR = refineV[PackageId.ValidVersion](msg.from) match {
+      case Left(e) => throw DeserializationException(RefinementError(msg.from, e))
+      case Right(v) => v
+    }
+    val toR = refineV[PackageId.ValidVersion](msg.from) match {
+      case Left(e) => throw DeserializationException(RefinementError(msg.from, e))
+      case Right(v) => v
+    }
```

Comment:
Same comment as above, should use `msg.to` but maybe better to change the event?